### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by tcharding**
In a continued effort to find my feet around here, and inspired by issue #128 I've done a codebase wide audit of the docs (primarily just rustdocs but I glanced at `//` docs as well). Each change is in a separate commit so can be removed if resistance is met. (_"resistance is futile"_).

I've based the stylistic decisions on [work done](https://github.com/rust-bitcoin/rust-bitcoin/pull/704) in rust-bitcoin. 

I believe the only controversial change is the last (commit: da161c9 Use rust-bitcoin module doc style), please review that one carefully.

### CODE REVIEW

**Comment by apoelstra** [comment]
concept ACK.

This PR touches a lot of code and is likely to get quickly out-of-date (and conversely to invalidate lots of open PRs). For that reason I'm hesitant to move forward with it, but I'm happy to try if others want to.

**Comment by tcharding** [comment]
Not sure what to do about this. I'll put it back to draft until I some idea comes to me.

Cheers.

**Comment by tcharding** [comment]
I'm reopening this because we have a bit of a quiet time right now, there are not many PRs open for this to conflict with, possibly only @dr-orlovsky's #309?

File: src/context.rs [hunk 1]
```diff
@@ -60,8 +60,8 @@ pub mod global {
 }
 
 
-/// A trait for all kinds of Context's that lets you define the exact flags and a function to deallocate memory.
-/// It shouldn't be possible to implement this for types outside this crate.
+/// A trait for all kinds of contexts that lets you define the exact flags and a function to
+/// deallocate memory. It shouldn't be possible to implement this for types outside this crate.
```
Comment (apoelstra) [inline src/context.rs:64]: In 269bde042f9d6bfccb857c6921e71f01a9854f35:

You could change "shouldn't be possible" to "isn't possible". I think we've come to trust rustc more than when this comment was written :)

File: src/context.rs [hunk 2]
```diff
@@ -98,8 +98,8 @@ pub struct AllPreallocated<'buf> {
 mod private {
     use super::*;
     // A trick to prevent users from implementing a trait.
-    // on one hand this trait is public, on the other it's in a private module
-    // so it's not visible to anyone besides it's parent (the context module)
+    // On one hand this trait is public, on the other it's in a private module
+    // so it's not visible to anyone besides it's parent (the context module).
```
Comment (apoelstra) [inline src/context.rs:102]: In 5e07e7596b3c128c2290089f87a2851946fb538d:

Should also change the second `it's` to `its`

Comment (apoelstra) [reply src/context.rs:102]: Also we could probably drop the exposition here; it's a pretty standard idiom now.

File: src/lib.rs [hunk 3]
```diff
@@ -414,9 +414,10 @@ impl<C: Context> Secp256k1<C> {
         (bytes + word_size - 1) / word_size
     }
 
-    /// (Re)randomizes the Secp256k1 context for cheap sidechannel resistance;
-    /// see comment in libsecp256k1 commit d2275795f by Gregory Maxwell. Requires
-    /// compilation with "rand" feature.
+    /// (Re)randomizes the Secp256k1 context for cheap sidechannel resistance.
```
Comment (apoelstra) [inline src/lib.rs:416]: In 3fa67624375ba529ad37f7786ca759224ff139fd:

Let's replace "cheap" with "extra". The cost is comparable to a signing operation.

**Comment by apoelstra** [comment]
Done reviewing c9e6ca168000aa3b0aea128048a7f7a40986398c

I think it's a quiet enough time that we can get this in, it's not *that* disruptive and we're mostly done with Schnorr signatures.

**Comment by tcharding** [comment]
All suggestions implemented as separate additional patches, thanks.

**Pull request merged**